contracts/Market.sol

```diff
@@ -870,6 +938,177 @@ contract Market is
+    return (countOfRemovalsAllocated, ids, amounts, suppliers);
+  }
+
+  /**
+   * @notice Prepares an order from a specific supplier to be fulfilled.
+   * @dev This function is responsible for calculating the certificate amount, validating and allocating the supply
+   * from a specific supplier, and permitting the bpNori transfer.
+   * @param certificateAmount The total amount for the certificate.
+   * @param supplier The only supplier address from which to purchase carbon removals in this transaction.
+   * @param deadline The EIP2612 permit deadline in Unix time.
+   * @param v The recovery identifier for the permit's secp256k1 signature
+   * @param r The r value for the permit's secp256k1 signature
+   * @param s The s value for the permit's secp256k1 signature
+   * @return countOfRemovalsAllocated The number of distinct removal IDs used to fulfill this order.
+   * @return ids An array of the removal IDs being drawn from to fulfill this order.
+   * @return amounts An array of amounts being allocated from each corresponding removal token.
+   * @return suppliers The address of the supplier who owns each corresponding removal token.
+   */
+  function _prepareSwapFromSupplier(
+    uint256 certificateAmount,
+    address supplier,
+    uint256 deadline,
+    uint8 v,
+    bytes32 r,
+    bytes32 s
+  )
+    internal
+    returns (
+      uint256 countOfRemovalsAllocated,
+      uint256[] memory ids,
+      uint256[] memory amounts,
+      address[] memory suppliers
+    )
+  {
+    _validatePrioritySupply({
+      certificateAmount: certificateAmount,
+      availableSupply: _removal.getMarketBalance()
+    });
+    (countOfRemovalsAllocated, ids, amounts) = _allocateSupplySingleSupplier({
+      certificateAmount: certificateAmount,
+      supplier: supplier
+    });
+    suppliers = new address[](countOfRemovalsAllocated).fill({val: supplier});
+    _bridgedPolygonNORI.permit({
+      owner: _msgSender(),
+      spender: address(this),
+      value: certificateAmount,
+      deadline: deadline,
+      v: v,
+      r: r,
+      s: s
+    });
+    return (countOfRemovalsAllocated, ids, amounts, suppliers);
+  }
+
+  /**
+   * @notice Fulfill an order without charging the Nori fee.
+   * @dev This function is responsible for paying suppliers, routeing tokens to the RestrictedNORI contract, updating
+   * accounting, and minting the Certificate.
+   * @param certificateAmount The total amount for the certificate.
+   * @param operator The message sender.
+   * @param recipient The recipient of the certificate.
+   * @param countOfRemovalsAllocated The number of distinct removal IDs that are involved in fulfilling this order.
+   * @param ids An array of removal IDs involved in fulfilling this order.
+   * @param amounts An array of amounts being allocated from each corresponding removal token.
+   * @param suppliers An array of suppliers.
+   */
+  function _fulfillOrderWithoutFee(
+    uint256 certificateAmount,
+    address operator,
```

Comment:
Is it possible to rename this to `from`? `operator` typically means the address which is an approved spender of the tokens, but this looks like it's using `from: operator` yet operator == `msg.sender` which is confusing (and potentially a misuse of operator params if it's actually relying on msg.sender being the operator somewhere-- which I didn't see but maybe i didn't look close enough)